src/robot/adapter/socket/socket_adapter.py

```diff
@@ -36,33 +38,45 @@ def __init__(self, reader: StreamReader, writer: StreamWriter):
             msg_str = msg.decode()
             raw_payload = binascii.unhexlify(msg_str)
 
+            self.packet_count_recv += 1
             LOGGER.get().debug('socket_can_adapter_received',
-                               raw_payload=raw_payload)
+                               raw_payload=raw_payload,
+                               packet_number_recv=self.packet_count_recv)
 
             for handler in self.handlers:
                 await handler(raw_payload)
 
     async def send(self, data: bytes) -> None:
-        self.writer.write(data)
+        hex_payload = binascii.hexlify(data)
+        self.packet_count_sent += 1
```

Comment:
maybe only count the packet as sent after the end of transmission? (in case it fails)
Do you manage socket write errors?

Reply:
+1

Reply:
After the drain

Reply:
Will add counters in another commit